tests/lib/rules/named-functions-in-promises.js

```diff
@@ -49,23 +73,50 @@ eslintTester.run('named-functions-in-promises', rule, {
         message: 'Use named functions defined on objects to handle promises',
       }],
     }, {
-      code: 'user.save().then(() => user.reload());',
+      code: 'user.save().catch(() => {return error.handle();});',
       parserOptions: { ecmaVersion: 6 },
       errors: [{
         message: 'Use named functions defined on objects to handle promises',
       }],
     }, {
-      code: 'user.save().catch(() => {return error.handle();});',
+      code: 'user.save().finally(() => {return finallyDo();});',
```

Comment:
I'd add one or three more tests like this, but with `allowConciseArrow` set to `true` just to be totally sure it works like expected :) Of course it should fail then too, because of the arrow function's body not being a CallExpression.